src/libcmd/installables.cc

```diff
@@ -491,7 +491,11 @@ Installables SourceExprCommand::parseInstallables(
             throw UsageError("'--file' and '--expr' are exclusive");
 
         // FIXME: backward compatibility hack
-        if (file) evalSettings.pureEval = false;
+        if (file) {
+            if (evalSettings.pureEval && evalSettings.pureEval.overridden)
+                throw UsageError("'--file' is not compatible with '--pure-eval'");
```

Comment:
It can actually work

```
nix build --pure-eval --file . -I . hello-unfree
```

So I am unsure if we should break it?

Reply:
With this PR it shows:
```
$ nix build --pure-eval --file . -I . hello-unfree
error: '--file' is not compatible with '--pure-eval'
Try 'nix --help' for more information.
```

It previously worked because the code above overwrote the user's `pureEval` setting.